Shocksupershaman - Copy/src/main/java/com/example/shocksupershaman/shocktask1/UserController.java

```diff
@@ -0,0 +1,45 @@
+package com.example.shocksupershaman.shocktask1;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+import java.util.List;
+import java.util.Optional;
+
+@RestController
+public class UserController {
+
+    @Autowired
+    private UserService userService;
+
+    @GetMapping("/getAllUsers")
+    public ResponseEntity<List<UserDTO>> getAllUsers() {
+        List<UserDTO> userDTOList = userService.getAllUsers();
+        return ResponseEntity.ok(userDTOList);
+    }
+
+    @GetMapping("/getUserById/{id}")

```

Comment:
Same. Path should be /{id}. After you add /users on top of the controller - final path will look like /users/{id} for get by id endpoint